src/numa.h

```diff
@@ -139,6 +140,54 @@ class NumaReplicatedAccessToken {
 // Until Stockfish doesn't support exceptions all places where an exception should be thrown
 // are replaced by std::exit.
 class NumaConfig {
+   private:
+#if defined(_WIN64)
+    struct WindowsAffinity {
+        std::optional<std::set<CpuIndex>> oldApi;
+        std::optional<std::set<CpuIndex>> newApi;
+        bool                              isDeterminate = true;
+
+        std::optional<std::set<CpuIndex>> get_combined() const {
+            if (!oldApi.has_value())
+                return newApi;
+            if (!newApi.has_value())
+                return oldApi;
+
+            std::set<CpuIndex> intersect;
+            std::set_intersection(oldApi->begin(), oldApi->end(), newApi->begin(), newApi->end(),
+                                  std::inserter(intersect, intersect.begin()));
+            return intersect;
+        }
+    };
+
+    // Since Windows 11 and Windows Server 2022 thread affinities can span
+    // processor groups and can be set as such by a new WinAPI function.
+    // However, we may need to force using the old API if we detect
+    // that the process has affinity set by the old API already and we want to override that.
+    static bool use_old_affinity_api() {
+        HMODULE k32                            = GetModuleHandle(TEXT("Kernel32.dll"));
+        auto    SetThreadSelectedCpuSetMasks_f = SetThreadSelectedCpuSetMasks_t(
+          (void (*)()) GetProcAddress(k32, "SetThreadSelectedCpuSetMasks"));
+
+        if (SetThreadSelectedCpuSetMasks_f == nullptr)
+            return true;
+
+        auto GetProcessGroupAffinity_f =
+          GetProcessGroupAffinity_t((void (*)()) GetProcAddress(k32, "GetProcessGroupAffinity"));
+
+        USHORT            GroupCount = 1;
+        alignas(4) USHORT GroupArray[1];
```

Comment:
maybe just move this grouparray part in a function together with the later usage, so that we dont accidentially remove the alignas here and dont have to duplicate the comment?

Reply:
alright, this was a good call, because while I was doing this I realized that we can simplify a bit which functions we resolve dynamically because we set
```
#define _WIN32_WINNT 0x0601
```

Also realized the current `static` functions we had in `NumaConfig` were basically similar wrappers over system API, so I moved them outside of `NumaConfig` too.

Reply:
a large chunk of `bind_current_thread_to_numa_node` could also be extracted outside in the future